File: leet_0240_search_matrix_II___diagonal_traversal.cc
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target)
    {
        int c = matrix[0].size();
        int r = matrix.size();
        int j = c - 1;
        int i = 0;
        while (i < r && j > -1)
        {
            int val = matrix[i][j];
            if (val == target)
                return true ;
            else if (val < target)
                ++ i;
            else
                -- j;
        }
        return false ;
    }
};

File: leet_0240_search_matrix_II___diagonal_traversal.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */

var searchMatrix = function(matrix, target) {
    let j = matrix[0].length - 1
    let r = matrix.length
    let i = 0
    while (i < r && j > -1){
        let val = matrix[i][j]
        if (target == val)
            return true
        else if (target > val)
            ++ i
        else
            -- j
    }
    return false
};
